In [1]:
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## The Orchard BI Prompt
The NYC Restaurant Task
Assume you have a friend who loves Thai food, but refuses to eat at any place with less than a B rating
by the Health Department.
1. Using the language and data store of your choice, create an ETL (Extract, Transform & Load)
job to ingest this ~500k rows DOHMH New York City Restaurant Inspection Results data set
from NYC Open Data (Download Link:
https://data.cityofnewyork.us/api/views/43nn-pn8j/rows.csv?accessType=DOWNLOAD.)
2. Using your data store, generate a list of the top 10 Thai restaurants that meet your friend's
criteria. You could simply provide a SQL query to answer this and export it into a data viz
tool, but it be amazing if you could build a web frontend to answer the question.
3. Create a data viz or two showing the results of the question.
4. Please publish your code on Github. [OPTIONAL] Bonus points if you deploy your app to a free
service like Heroku, Google App Engine, or Flask so that we can see it in action.

In [2]:
raw_data = pd.read_csv('DOHMH_New_York_City_Restaurant_Inspection_Results.csv',usecols=['CAMIS','DBA','BORO','ZIPCODE','CUISINE DESCRIPTION','INSPECTION DATE','SCORE','GRADE','GRADE DATE'], encoding="utf-8")
df = pd.DataFrame(raw_data)

df.head()

CAMIS                         DBA       BORO  ZIPCODE  \
0  40680764                 JOE'S PLACE      BRONX  10472.0   
1  50068094          MI VISABUELO CHIMI   BROOKLYN  11208.0   
2  50008826      RAY'S PIZZA BAGEL CAFE  MANHATTAN  10003.0   
3  41258116                  MCDONALD'S   BROOKLYN  11229.0   
4  50009898  EL GRAN MALECON RESTAURANT   BROOKLYN  11237.0   

                                 CUISINE DESCRIPTION INSPECTION DATE  SCORE  \
0  Latin (Cuban, Dominican, Puerto Rican, South &...      06/08/2016   34.0   
1                                            Spanish      09/12/2017   12.0   
2                                      Pizza/Italian      02/11/2016    9.0   
3                                           American      03/21/2018   10.0   
4                                            Spanish      08/28/2018   59.0   

  GRADE  GRADE DATE  
0   NaN         NaN  
1     A  09/12/2017  
2     A  02/11/2016  
3     A  03/21/2018  
4   NaN         NaN

In [3]:
df.shape

(385239, 9)

In [4]:
useful = df.loc[(df['CUISINE DESCRIPTION'] == 'Thai') & df['GRADE'].isin(['A','B'])]

In [5]:
useful.head()

CAMIS                   DBA       BORO  ZIPCODE CUISINE DESCRIPTION  \
37   40972155  CHAI THAI RESTAURANT   BROOKLYN  11249.0                Thai   
291  50046126                DANNEE   BROOKLYN  11205.0                Thai   
402  50011657                 SPICE  MANHATTAN  10003.0                Thai   
505  41083574     LAND THAI KITCHEN  MANHATTAN  10024.0                Thai   
567  50004740             THAI CAFE   BROOKLYN  11222.0                Thai   

    INSPECTION DATE  SCORE GRADE  GRADE DATE  
37       07/18/2016   12.0     A  07/18/2016  
291      06/07/2017    9.0     A  06/07/2017  
402      09/17/2015   12.0     A  09/17/2015  
505      12/22/2015   14.0     B  12/22/2015  
567      05/03/2017   10.0     A  05/03/2017

In [6]:
df = useful.reset_index(drop=True)

In [7]:
df.head()

CAMIS                   DBA       BORO  ZIPCODE CUISINE DESCRIPTION  \
0  40972155  CHAI THAI RESTAURANT   BROOKLYN  11249.0                Thai   
1  50046126                DANNEE   BROOKLYN  11205.0                Thai   
2  50011657                 SPICE  MANHATTAN  10003.0                Thai   
3  41083574     LAND THAI KITCHEN  MANHATTAN  10024.0                Thai   
4  50004740             THAI CAFE   BROOKLYN  11222.0                Thai   

  INSPECTION DATE  SCORE GRADE  GRADE DATE  
0      07/18/2016   12.0     A  07/18/2016  
1      06/07/2017    9.0     A  06/07/2017  
2      09/17/2015   12.0     A  09/17/2015  
3      12/22/2015   14.0     B  12/22/2015  
4      05/03/2017   10.0     A  05/03/2017

In [8]:
df.dtypes

CAMIS                    int64
DBA                     object
BORO                    object
ZIPCODE                float64
CUISINE DESCRIPTION     object
INSPECTION DATE         object
SCORE                  float64
GRADE                   object
GRADE DATE              object
dtype: object

In [9]:
df.sort_values(['GRADE DATE'], ascending=[False], inplace=True)

In [10]:
df.head()

CAMIS             DBA       BORO  ZIPCODE CUISINE DESCRIPTION  \
1125  41716975       DER KRUNG  MANHATTAN  10019.0                Thai   
1593  41716975       DER KRUNG  MANHATTAN  10019.0                Thai   
310   50033872  ESAN THAI FOOD     QUEENS  11377.0                Thai   
1416  50033872  ESAN THAI FOOD     QUEENS  11377.0                Thai   
1661  50000496        POTJANEE  MANHATTAN  10014.0                Thai   

     INSPECTION DATE  SCORE GRADE  GRADE DATE  
1125      12/30/2015    9.0     A  12/30/2015  
1593      12/30/2015    9.0     A  12/30/2015  
310       12/29/2016   11.0     A  12/29/2016  
1416      12/29/2016   11.0     A  12/29/2016  
1661      12/29/2016    9.0     A  12/29/2016

In [11]:
df.isnull().values.any()

True

In [12]:
df.isnull().any()

CAMIS                  False
DBA                    False
BORO                   False
ZIPCODE                 True
CUISINE DESCRIPTION    False
INSPECTION DATE        False
SCORE                  False
GRADE                  False
GRADE DATE             False
dtype: bool

In [13]:
df.ZIPCODE.mean()

10633.373814041746

In [14]:
df.fillna(df.ZIPCODE.mean())

CAMIS                              DBA           BORO  ZIPCODE  \
1125  41716975                        DER KRUNG      MANHATTAN  10019.0   
1593  41716975                        DER KRUNG      MANHATTAN  10019.0   
310   50033872                   ESAN THAI FOOD         QUEENS  11377.0   
1416  50033872                   ESAN THAI FOOD         QUEENS  11377.0   
1661  50000496                         POTJANEE      MANHATTAN  10014.0   
2054  50033872                   ESAN THAI FOOD         QUEENS  11377.0   
2112  50000496                         POTJANEE      MANHATTAN  10014.0   
1436  41156192                          LANTERN       BROOKLYN  11201.0   
1814  41156192                          LANTERN       BROOKLYN  11201.0   
382   41720477            AROY DEE THAI KITCHEN      MANHATTAN  10038.0   
888   41720477            AROY DEE THAI KITCHEN      MANHATTAN  10038.0   
421   41004486             GALANGA THAI COOKING      MANHATTAN  10012.0   
1565  41004486             GALANGA THAI COOKING      MANHATTAN  10012.0   
1600  50004825  THE OLD SIAM RESTAURANT AND BAR         QUEENS  11375.0   
1770  50004825  THE OLD SIAM RESTAURANT AND BAR         QUEENS  11375.0   
1522  41455608                        BORN THAI       BROOKLYN  11217.0   
317   41455608                        BORN THAI       BROOKLYN  11217.0   
580   41455608                        BORN THAI       BROOKLYN  11217.0   
1138  50000950                         THAI 101         QUEENS  11361.0   
1843  50007431                        KHAO KANG         QUEENS  11373.0   
379   41083574                LAND THAI KITCHEN      MANHATTAN  10024.0   
288   50000950                         THAI 101         QUEENS  11361.0   
1673  50007431                        KHAO KANG         QUEENS  11373.0   
1982  50000950                         THAI 101         QUEENS  11361.0   
867   41083574                LAND THAI KITCHEN      MANHATTAN  10024.0   
3     41083574                LAND THAI KITCHEN      MANHATTAN  10024.0   
498   50069543                         TOP THAI      MANHATTAN  10012.0   
2059  50069543                         TOP THAI      MANHATTAN  10012.0   
1797  41591834                             NGAM      MANHATTAN  10003.0   
1182  41591834                             NGAM      MANHATTAN  10003.0   
...        ...                              ...            ...      ...   
1351  50009573                   SONG KRAN THAI      MANHATTAN  10001.0   
865   50009573                   SONG KRAN THAI      MANHATTAN  10001.0   
363   50009573                   SONG KRAN THAI      MANHATTAN  10001.0   
1641  41689402                        KARE THAI      MANHATTAN  10019.0   
1086  41590317                        THAI ROCK         QUEENS  11693.0   
586   41590317                        THAI ROCK         QUEENS  11693.0   
610   40904478                       LOVELY DAY      MANHATTAN  10012.0   
1106  40904478                       LOVELY DAY      MANHATTAN  10012.0   
969   40904478                       LOVELY DAY      MANHATTAN  10012.0   
862   50011155                       TOMMY THAI         QUEENS  11373.0   
271   50011155                       TOMMY THAI         QUEENS  11373.0   
309   50011155                       TOMMY THAI         QUEENS  11373.0   
163   41431917              FOREST THAI CUISINE  STATEN ISLAND  10310.0   
466   41431917              FOREST THAI CUISINE  STATEN ISLAND  10310.0   
1702  40557279               BENNIE'S THAI CAFE      MANHATTAN  10038.0   
1697  40557279               BENNIE'S THAI CAFE      MANHATTAN  10038.0   
2044  40557279               BENNIE'S THAI CAFE      MANHATTAN  10038.0   
1498  41659574                         THE NUAA      MANHATTAN  10065.0   
1340  41659574                         THE NUAA      MANHATTAN  10065.0   
527   41659574                         THE NUAA      MANHATTAN  10065.0   
772   41659574                         THE NUAA      MANHATTAN  10065.0   
2111  40392724   JAIYA THAI ORIENT

In [15]:
print df

         CAMIS                              DBA           BORO  ZIPCODE  \
1125  41716975                        DER KRUNG      MANHATTAN  10019.0   
1593  41716975                        DER KRUNG      MANHATTAN  10019.0   
310   50033872                   ESAN THAI FOOD         QUEENS  11377.0   
1416  50033872                   ESAN THAI FOOD         QUEENS  11377.0   
1661  50000496                         POTJANEE      MANHATTAN  10014.0   
2054  50033872                   ESAN THAI FOOD         QUEENS  11377.0   
2112  50000496                         POTJANEE      MANHATTAN  10014.0   
1436  41156192                          LANTERN       BROOKLYN  11201.0   
1814  41156192                          LANTERN       BROOKLYN  11201.0   
382   41720477            AROY DEE THAI KITCHEN      MANHATTAN  10038.0   
888   41720477            AROY DEE THAI KITCHEN      MANHATTAN  10038.0   
421   41004486             GALANGA THAI COOKING      MANHATTAN  10012.0   
1565  41004486           

In [16]:
df=df.dropna()

In [17]:
df.shape

(2108, 9)

In [18]:
df.isnull().any()

CAMIS                  False
DBA                    False
BORO                   False
ZIPCODE                False
CUISINE DESCRIPTION    False
INSPECTION DATE        False
SCORE                  False
GRADE                  False
GRADE DATE             False
dtype: bool

In [19]:
df.ZIPCODE = df.ZIPCODE.astype(int)

In [20]:
df.SCORE = df.SCORE.astype(int)

In [21]:
df['INSPECTION DATE'] =  pd.to_datetime(raw_data['INSPECTION DATE'])

In [22]:
df['GRADE DATE'] =  pd.to_datetime(raw_data['GRADE DATE'])

In [23]:
df.head()

CAMIS             DBA       BORO  ZIPCODE CUISINE DESCRIPTION  \
1125  41716975       DER KRUNG  MANHATTAN    10019                Thai   
1593  41716975       DER KRUNG  MANHATTAN    10019                Thai   
310   50033872  ESAN THAI FOOD     QUEENS    11377                Thai   
1416  50033872  ESAN THAI FOOD     QUEENS    11377                Thai   
1661  50000496        POTJANEE  MANHATTAN    10014                Thai   

     INSPECTION DATE  SCORE GRADE GRADE DATE  
1125      2015-01-12      9     A 2015-01-12  
1593      2017-05-05      9     A 2017-05-05  
310       2018-05-10     11     A        NaT  
1416      2018-07-02     11     A 2018-07-02  
1661      2017-08-18      9     A        NaT

In [24]:
df.isnull().any()

CAMIS                  False
DBA                    False
BORO                   False
ZIPCODE                False
CUISINE DESCRIPTION    False
INSPECTION DATE        False
SCORE                  False
GRADE                  False
GRADE DATE              True
dtype: bool

In [25]:
df = df.drop(columns='GRADE DATE')

In [26]:
df.sort_values(['CAMIS', 'INSPECTION DATE', 'GRADE', 'SCORE'], ascending=[True, False, True, True])

CAMIS                                DBA       BORO  ZIPCODE  \
1935  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
2083  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
2084  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1062  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1574  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
556   40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1049  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1240  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
655   40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1167  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1308  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1354  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
2099  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1331  40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
764   40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
127   40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
2111  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
272   40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
859   40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1110  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1772  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
280   40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1657  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1030  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1346  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1259  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1115  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1345  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1756  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
1015  40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
...        ...                                ...        ...      ...   
1952  50082311                          KATI SHOP  MANHATTAN    10022   
1219  50082385                  THAI DIRECT BOWLS  MANHATTAN    10009   
500   50083010                             MEKONG  MANHATTAN    10025   
297   50083123                     MAISON BANGKOK  MANHATTAN    10075   
1851  50083123                     MAISON BANGKOK  MANHATTAN    10075   
878   50083836                   PHO GET ABOUT MI  MANHATTAN    10014   
1104  50083836                   PHO GET ABOUT MI  MANHATTAN    10014   
906   50084162                         Asomm Thai     QUEENS    11385   
1738  50084162                         Asomm Thai     QUEENS    11385   
133   50084400                  Noree Thai Bazaar  MANHATTAN    10002   
595   50084400                  Noree Thai Bazaar  MANHATTAN    10002   
1734  50084401                            JAI DEE   BROOKLYN    11215   
1068  50084401                            JAI DEE   BROOKLYN    11215   
1672  50084401                            JAI DEE   BROOKLYN    11215   
614   50084430                     THAI V EXPRESS  MANHATTAN    10010   
1276  50084430                     THAI V EXPRESS  MANHATTAN    10010   
624   50084554                         STANG THAI   BROOKLYN    11218   
622   50084554                         STANG THAI   BROOKLYN    11218   
2028  50084587  HOO JIA TAIWANESE CUISINE (ST#26)     QUEENS    11354   
2065  50084746          WILLIAMSBURG THAI CUISINE   BROOKLYN    11249   
7     50084746          WILLIAMSBURG THAI CUISINE   BROOKLYN    11249   
1804  50084746          WILLIAMSBURG THAI CUISINE   BROOKLYN    11249   
1032  50084860                            THAI ME   BROOKLYN    112

In [27]:
df.drop_duplicates(keep=False, inplace=True)

In [28]:
df.shape

(2106, 8)

In [29]:
df= df.reset_index(drop=True)

To write this in a SQL it would look something like this:
```SELECT DISTINCT DOHMH_New_York_City_Restaurant_Inspection_Results."CAMIS", 
    DOHMH_New_York_City_Restaurant_Inspection_Results."DBA",
    DOHMH_New_York_City_Restaurant_Inspection_Results."BORO",
    cast(DOHMH_New_York_City_Restaurant_Inspection_Results."ZIPCODE" as int),
    DOHMH_New_York_City_Restaurant_Inspection_Results."CUISINE DESCRIPTION",
    DOHMH_New_York_City_Restaurant_Inspection_Results."INSPECTION DATE",
    cast(DOHMH_New_York_City_Restaurant_Inspection_Results."SCORE" as int),
    DOHMH_New_York_City_Restaurant_Inspection_Results."GRADE"
   FROM DOHMH_New_York_City_Restaurant_Inspection_Results
   WHERE DOHMH_New_York_City_Restaurant_Inspection_Results."CUISINE DESCRIPTION" == 'Thai' 
   AND DOHMH_New_York_City_Restaurant_Inspection_Results."GRADE" IN ('A','B') 
   AND DOHMH_New_York_City_Restaurant_Inspection_Results."ZIPCODE" IS NOT NULL
   GROUP BY DOHMH_New_York_City_Restaurant_Inspection_Results."BORO"
   ORDER BY DOHMH_New_York_City_Restaurant_Inspection_Results."GRADE" ASC,
   DOHMH_New_York_City_Restaurant_Inspection_Results."SCORE" ASC;
   ```

In [30]:
df = df.sort_values(['CAMIS', 'INSPECTION DATE', 'GRADE'], ascending=[True, False, True])

In [31]:
df= df.reset_index(drop=True)

In [32]:
df.head()

CAMIS                      DBA       BORO  ZIPCODE CUISINE DESCRIPTION  \
0  40374088  PONGSRI THAI RESTAURANT  MANHATTAN    10036                Thai   
1  40374088  PONGSRI THAI RESTAURANT  MANHATTAN    10036                Thai   
2  40374088  PONGSRI THAI RESTAURANT  MANHATTAN    10036                Thai   
3  40374088  PONGSRI THAI RESTAURANT  MANHATTAN    10036                Thai   
4  40374088  PONGSRI THAI RESTAURANT  MANHATTAN    10036                Thai   

  INSPECTION DATE  SCORE GRADE  
0      2018-10-18     20     B  
1      2018-09-24     20     B  
2      2018-05-24     20     B  
3      2018-04-24     24     B  
4      2018-01-22     13     A

In [33]:
df.drop_duplicates(subset=['CAMIS','DBA'],keep="first", inplace=True)

In [34]:
df.head()

CAMIS                             DBA       BORO  ZIPCODE  \
0   40374088         PONGSRI THAI RESTAURANT  MANHATTAN    10036   
16  40392724  JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
34  40557279              BENNIE'S THAI CAFE  MANHATTAN    10038   
48  40704155             ERAWAN THAI CUISINE     QUEENS    11361   
56  40799210                            JOYA   BROOKLYN    11201   

   CUISINE DESCRIPTION INSPECTION DATE  SCORE GRADE  
0                 Thai      2018-10-18     20     B  
16                Thai      2018-11-17     13     A  
34                Thai      2018-12-15     10     A  
48                Thai      2018-12-11      4     A  
56                Thai      2018-08-28     13     A

In [35]:
df.shape

(285, 8)

In [36]:
df

CAMIS                                DBA       BORO  ZIPCODE  \
0     40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
16    40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
34    40557279                 BENNIE'S THAI CAFE  MANHATTAN    10038   
48    40704155                ERAWAN THAI CUISINE     QUEENS    11361   
56    40799210                               JOYA   BROOKLYN    11201   
63    40880867                   OTT THAI CUISINE   BROOKLYN    11222   
72    40904478                         LOVELY DAY  MANHATTAN    10012   
81    40957398                           KUMA INN  MANHATTAN    10002   
86    40968682                       STAR OF SIAM     QUEENS    11375   
95    40972155               CHAI THAI RESTAURANT   BROOKLYN    11249   
109   41003679                 TEEDA THAI CUISINE   BROOKLYN    11231   
116   41004486               GALANGA THAI COOKING  MANHATTAN    10012   
123   41083574                  LAND THAI KITCHEN  MANHATTAN    10024   
130   41112819                               SONG   BROOKLYN    11215   
135   41156192                            LANTERN   BROOKLYN    11201   
147   41163809             LUCKY THAILAND KITCHEN     QUEENS    11421   
154   41171296                   KIIN THAI EATERY  MANHATTAN    10003   
165   41193976           THAI ELEPHANT RESTAURANT     QUEENS    11105   
178   41224979                         SABAY THAI     QUEENS    11373   
195   41227184                        STICKY RICE  MANHATTAN    10002   
204   41230254               BKNY THAI RESTAURANT     QUEENS    11361   
213   41241757                        THAI MARKET  MANHATTAN    10025   
221   41248845                         CAFE CHILI   BROOKLYN    11201   
231   41275701                         TO BE THAI   BROOKLYN    11218   
235   41300979              HONEY'S THAI PAVILION      BRONX    10461   
243   41303343          LENG AUTHENTIC THAI ASIAN     QUEENS    11106   
250   41310519                       LANTERN THAI  MANHATTAN    10003   
262   41321253           BOON CHU THAI RESTAURANT     QUEENS    11373   
270   41321677             THAI GRILL & SUSHI BAR  MANHATTAN    10017   
281   41345907                   MEE THAI CUISINE   BROOKLYN    11217   
...        ...                                ...        ...      ...   
2033  50075231                          CHAO THAI     QUEENS    11373   
2034  50075334                   LEMONGRASS GRILL   BROOKLYN    11201   
2037  50075350                  PRIK THAI KITHCEN     QUEENS    11103   
2039  50076096                      SPICY SHALLOT     QUEENS    11373   
2045  50076740                VIV THAI RESTAURANT  MANHATTAN    10019   
2047  50076867                      DAUNGDEE THAI   BROOKLYN    11211   
2049  50076881           AT NINE RESTAURANT & BAR  MANHATTAN    10036   
2055  50077461                  THAI DIVA CUISINE     QUEENS    11377   
2059  50077505                              KIN'D     QUEENS    11366   
2062  50077758                  THAI NOODLE HOUSE  MANHATTAN    10019   
2063  50078693                        Zenith Thai     QUEENS    11385   
2067  50079810                          ANA RAMEN  MANHATTAN    10019   
2071  50081677                          SALA THAI  MANHATTAN    10023   
2072  50081713              Tam Sang Thai Kitchen  MANHATTAN    10016   
2075  50082311                          KATI SHOP  MANHATTAN    10022   
2077  50082385                  THAI DIRECT BOWLS  MANHATTAN    10009   
2078  50083010                             MEKONG  MANHATTAN    10025   
2079  50083123                     MAISON BANGKOK  MANHATTAN    10075   
2081  50083836                   PHO GET ABOUT MI  MANHATTAN    10014   
2083  50084162                         Asomm Thai     QUEENS    11385   
2085  50084400                  Noree Thai Bazaar  MANHATTAN    10002   
2087  50084401                            JAI DEE   BROOKLYN    11215   
2090  50084430                     THAI V EXPRESS  MANHATTAN    100

In [37]:
df= df.reset_index(drop=True)

In [38]:
df

CAMIS                                DBA       BORO  ZIPCODE  \
0    40374088            PONGSRI THAI RESTAURANT  MANHATTAN    10036   
1    40392724     JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016   
2    40557279                 BENNIE'S THAI CAFE  MANHATTAN    10038   
3    40704155                ERAWAN THAI CUISINE     QUEENS    11361   
4    40799210                               JOYA   BROOKLYN    11201   
5    40880867                   OTT THAI CUISINE   BROOKLYN    11222   
6    40904478                         LOVELY DAY  MANHATTAN    10012   
7    40957398                           KUMA INN  MANHATTAN    10002   
8    40968682                       STAR OF SIAM     QUEENS    11375   
9    40972155               CHAI THAI RESTAURANT   BROOKLYN    11249   
10   41003679                 TEEDA THAI CUISINE   BROOKLYN    11231   
11   41004486               GALANGA THAI COOKING  MANHATTAN    10012   
12   41083574                  LAND THAI KITCHEN  MANHATTAN    10024   
13   41112819                               SONG   BROOKLYN    11215   
14   41156192                            LANTERN   BROOKLYN    11201   
15   41163809             LUCKY THAILAND KITCHEN     QUEENS    11421   
16   41171296                   KIIN THAI EATERY  MANHATTAN    10003   
17   41193976           THAI ELEPHANT RESTAURANT     QUEENS    11105   
18   41224979                         SABAY THAI     QUEENS    11373   
19   41227184                        STICKY RICE  MANHATTAN    10002   
20   41230254               BKNY THAI RESTAURANT     QUEENS    11361   
21   41241757                        THAI MARKET  MANHATTAN    10025   
22   41248845                         CAFE CHILI   BROOKLYN    11201   
23   41275701                         TO BE THAI   BROOKLYN    11218   
24   41300979              HONEY'S THAI PAVILION      BRONX    10461   
25   41303343          LENG AUTHENTIC THAI ASIAN     QUEENS    11106   
26   41310519                       LANTERN THAI  MANHATTAN    10003   
27   41321253           BOON CHU THAI RESTAURANT     QUEENS    11373   
28   41321677             THAI GRILL & SUSHI BAR  MANHATTAN    10017   
29   41345907                   MEE THAI CUISINE   BROOKLYN    11217   
..        ...                                ...        ...      ...   
255  50075231                          CHAO THAI     QUEENS    11373   
256  50075334                   LEMONGRASS GRILL   BROOKLYN    11201   
257  50075350                  PRIK THAI KITHCEN     QUEENS    11103   
258  50076096                      SPICY SHALLOT     QUEENS    11373   
259  50076740                VIV THAI RESTAURANT  MANHATTAN    10019   
260  50076867                      DAUNGDEE THAI   BROOKLYN    11211   
261  50076881           AT NINE RESTAURANT & BAR  MANHATTAN    10036   
262  50077461                  THAI DIVA CUISINE     QUEENS    11377   
263  50077505                              KIN'D     QUEENS    11366   
264  50077758                  THAI NOODLE HOUSE  MANHATTAN    10019   
265  50078693                        Zenith Thai     QUEENS    11385   
266  50079810                          ANA RAMEN  MANHATTAN    10019   
267  50081677                          SALA THAI  MANHATTAN    10023   
268  50081713              Tam Sang Thai Kitchen  MANHATTAN    10016   
269  50082311                          KATI SHOP  MANHATTAN    10022   
270  50082385                  THAI DIRECT BOWLS  MANHATTAN    10009   
271  50083010                             MEKONG  MANHATTAN    10025   
272  50083123                     MAISON BANGKOK  MANHATTAN    10075   
273  50083836                   PHO GET ABOUT MI  MANHATTAN    10014   
274  50084162                         Asomm Thai     QUEENS    11385   
275  50084400                  Noree Thai Bazaar  MANHATTAN    10002   
276  50084401                            JAI DEE   BROOKLYN    11215   
277  50084430                     THAI V EXPRESS  MANHATTAN    10010   
278  50084554                         STANG THAI

In [39]:
df = df.drop(columns='CAMIS')

In [41]:
df = df.drop(columns='CUISINE DESCRIPTION')

In [42]:
df = df.drop(columns='INSPECTION DATE')

In [44]:
df

DBA       BORO  ZIPCODE  SCORE GRADE
0              PONGSRI THAI RESTAURANT  MANHATTAN    10036     20     B
1       JAIYA THAI ORIENTAL RESTAURANT  MANHATTAN    10016     13     A
2                   BENNIE'S THAI CAFE  MANHATTAN    10038     10     A
3                  ERAWAN THAI CUISINE     QUEENS    11361      4     A
4                                 JOYA   BROOKLYN    11201     13     A
5                     OTT THAI CUISINE   BROOKLYN    11222     10     A
6                           LOVELY DAY  MANHATTAN    10012     10     A
7                             KUMA INN  MANHATTAN    10002     12     A
8                         STAR OF SIAM     QUEENS    11375     12     A
9                 CHAI THAI RESTAURANT   BROOKLYN    11249     21     B
10                  TEEDA THAI CUISINE   BROOKLYN    11231     12     A
11                GALANGA THAI COOKING  MANHATTAN    10012      7     A
12                   LAND THAI KITCHEN  MANHATTAN    10024      8     A
13                                SONG   BROOKLYN    11215     12     A
14                             LANTERN   BROOKLYN    11201     10     A
15              LUCKY THAILAND KITCHEN     QUEENS    11421      7     A
16                    KIIN THAI EATERY  MANHATTAN    10003     13     A
17            THAI ELEPHANT RESTAURANT     QUEENS    11105     25     B
18                          SABAY THAI     QUEENS    11373     10     A
19                         STICKY RICE  MANHATTAN    10002     12     A
20                BKNY THAI RESTAURANT     QUEENS    11361     10     A
21                         THAI MARKET  MANHATTAN    10025     10     A
22                          CAFE CHILI   BROOKLYN    11201     11     A
23                          TO BE THAI   BROOKLYN    11218      9     A
24               HONEY'S THAI PAVILION      BRONX    10461      9     A
25           LENG AUTHENTIC THAI ASIAN     QUEENS    11106     12     A
26                        LANTERN THAI  MANHATTAN    10003     13     A
27            BOON CHU THAI RESTAURANT     QUEENS    11373      9     A
28              THAI GRILL & SUSHI BAR  MANHATTAN    10017     12     A
29                    MEE THAI CUISINE   BROOKLYN    11217     14     B
..                                 ...        ...      ...    ...   ...
255                          CHAO THAI     QUEENS    11373      8     A
256                   LEMONGRASS GRILL   BROOKLYN    11201     12     A
257                  PRIK THAI KITHCEN     QUEENS    11103      8     A
258                      SPICY SHALLOT     QUEENS    11373     10     A
259                VIV THAI RESTAURANT  MANHATTAN    10019      8     A
260                      DAUNGDEE THAI   BROOKLYN    11211     10     A
261           AT NINE RESTAURANT & BAR  MANHATTAN    10036     22     B
262                  THAI DIVA CUISINE     QUEENS    11377     23     B
263                              KIN'D     QUEENS    11366     12     A
264                  THAI NOODLE HOUSE  MANHATTAN    10019      7     A
265                        Zenith Thai     QUEENS    11385     11     A
266                          ANA RAMEN  MANHATTAN    10019     13     A
267                          SALA THAI  MANHATTAN    10023      8     A
268              Tam Sang Thai Kitchen  MANHATTAN    10016     12     A
269                          KATI SHOP  MANHATTAN    10022     12     A
270                  THAI DIRECT BOWLS  MANHATTAN    10009      8     A
271                             MEKONG  MANHATTAN    10025      6     A
272                     MAISON BANGKOK  MANHATTAN    10075     10     A
273                   PHO GET ABOUT MI  MANHATTAN    10014     10     A
274                         Asomm Thai     QUEENS    11385      8     A
275                  Noree Thai Bazaar  MANHATTAN    10002      7     A
276                            JAI DEE   BROOKLYN    11215     11     A
277                     THAI V EXPRESS  MANHATTAN    10010     10     A
278                         STANG THAI   BROOKLYN    11218     12     A
279

In [46]:
DATASET = df.groupby(['BORO', 'GRADE']).agg(['count'])

In [47]:
DATASET

DBA ZIPCODE SCORE
                    count   count count
BORO          GRADE                    
BRONX         A         5       5     5
BROOKLYN      A        65      65    65
              B         7       7     7
MANHATTAN     A       109     109   109
              B        24      24    24
QUEENS        A        61      61    61
              B        11      11    11
STATEN ISLAND A         3       3     3

 From the cleaned data set we've been able to narrow down the full list to 285 restaurants of interest. We see the majority of them are in Manhattan, but that could be do to the density of population in that particular boro. Staten Island seems like a good idea for a Thai Cuisine restaruant crawl, or the Bronx for a five course progressive dinner. Interesting next steps would be to map it out by ZIPCODE to make a walking map for my Thai restaurant enthusiast friend. It would be a nice holiday gift!

In [49]:
df.to_csv('Thai_cuisine.csv', sep=',', encoding='utf-8')